web/src/render.tsx

```diff
@@ -17,11 +17,56 @@
 import './main.scss';
 import ReactDom from 'react-dom';
 import React from 'react';
+import { encrypt_string } from '../pkg';
 
 const hljs = require('highlight.js');
 (window as any).hljs = hljs;
 require('highlightjs-line-numbers.js');
 
+class PasteForm extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      value: "Sample text"
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({value: event.target.value});
+  }
+
+  handleSubmit(event) {
+    try {
+      encrypt_string(this.state.value);
+    } catch (e) {
+      console.error(e);
+    }
+
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <pre className='paste'>
+        <form class="hljs centered" onSubmit={this.handleSubmit}>
+          <textarea value={this.state.value} onChange={this.handleChange} />
+          <input type="submit" value="submit" />
+        </form>
+      </pre>
```

Comment:
Wait, huh? Do you need to hold state for this? I _swear_ you can access the textarea value from the event provided to handleSubmit

Also use the placeholder attribute for "Sample text" pls

Reply:
i think only onChange can get the textarea value
onSubmit is tied to the form so i think state is required